File: b2b-service/src/test/java/com/expedia/b2b/documentation/Documentation.java
package com.expedia.b2b.documentation;

import io.restassured.RestAssured;
import io.restassured.specification.RequestSpecification;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.autoconfigure.restdocs.AutoConfigureRestDocs;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.web.server.LocalServerPort;
import org.springframework.context.annotation.Import;
import org.springframework.restdocs.RestDocumentationExtension;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ExtendWith(RestDocumentationExtension.class) // 스니펫 생성을 위한 선언
@AutoConfigureRestDocs
public class Documentation {
    @LocalServerPort
    int port;

    protected RequestSpecification spec;

    @BeforeEach
    public void setUp() {
        RestAssured.port = port;
    }
}

File: platform-server/member-server/src/test/kotlin/org/clonecoder/member/documentation/Documentation.kt
package org.clonecoder.member.documentation

import io.restassured.RestAssured
import io.restassured.builder.RequestSpecBuilder
import io.restassured.specification.RequestSpecification
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.web.server.LocalServerPort
import org.springframework.restdocs.RestDocumentationContextProvider
import org.springframework.restdocs.RestDocumentationExtension
import org.springframework.restdocs.operation.preprocess.Preprocessors.prettyPrint
import org.springframework.restdocs.restassured3.RestAssuredRestDocumentation.documentationConfiguration
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.TestPropertySource


@TestPropertySource(properties = ["spring.config.location=classpath:application-test.yml"])
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ExtendWith(RestDocumentationExtension::class)
internal class Documentation {

    @LocalServerPort
    lateinit var port: String

    protected lateinit var spec: RequestSpecification

    @BeforeEach
    fun setUp(documentation: RestDocumentationContextProvider) {
        RestAssured.port = port.toInt()
        this.spec = RequestSpecBuilder()
            .addFilter(documentationConfiguration(documentation)
                    .operationPreprocessors()
                    .withRequestDefaults(prettyPrint())
                    .withResponseDefaults(prettyPrint()))
            .build()
    }
}